src/hal/src/pso/mod.rs

```diff
@@ -151,26 +151,52 @@ impl fmt::Display for Stage {
 }
 
 /// Shader entry point.
-#[derive(Debug, Copy)]
-pub struct EntryPoint<'a, B: Backend> {
+#[derive(Debug)]
+pub struct EntryPoint<'a, B: Backend, S = RawSpecialization<'a>> {
     /// Entry point name.
     pub entry: &'a str,
     /// Shader module reference.
     pub module: &'a B::ShaderModule,
     /// Specialization.
-    pub specialization: Specialization<'a>,
+    pub specialization: S,
+}
+
+impl<'a, B, S> EntryPoint<'a, B, S>
+where
+    B: Backend,
+    S: Specialization<'a>,
+{
+    /// Convert `specialization` of the `EntryPoint` into raw type
+    /// acceptable by pipeline creation functions.
+    pub fn into_raw(&self, storage: &'a mut SpecializationStorage) -> EntryPoint<'a, B> {
+        EntryPoint {
+            entry: self.entry,
+            module: self.module,
+            specialization: self.specialization.into_raw(storage),
+        }
+    }
 }
 
-impl<'a, B: Backend> Clone for EntryPoint<'a, B> {
+impl<'a, B, S> Clone for EntryPoint<'a, B, S>
+where
+    B: Backend,
+    S: Clone,
+{
     fn clone(&self) -> Self {
         EntryPoint {
             entry: self.entry,
             module: self.module,
-            specialization: self.specialization,
+            specialization: self.specialization.clone(),
         }
     }
 }
 
+impl<'a, B, S> Copy for EntryPoint<'a, B, S>
```

Comment:
why do you need `Copy`?

Reply:
I can `Copy` that's enough reason for impl

Reply:
I respectfully disagree. If the payload is big-ish, having the programmer to call `clone` can at least make the costs more explicit. Not all copies are good.

Reply:
I agree. Yet if `S` is big-ish it shall not implement `Copy` and then `EntryPoint` won't too.
And if `S` is small `EntryPoint` is small too.